src/sortStudents.ts

```diff
@@ -1,16 +1,66 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(students: Student[],
+  sortBy: SortType, order: SortOrder): Student[] {
+  const avgGradeMethod = (e1: Student, e2: Student): number => {
+    return e1.grades.reduce((sum, x): number => sum + x, 0)
+           / e1.grades.length - e2.grades.reduce((sum, x): number => sum + x, 0)
+           / e2.grades.length;
+  };
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  const CompareFn = (a: Student, b: Student): number => {
+    switch (true) {
```

Comment:
You should check the sortBy inside this switch/case construction. Make sure that you understand how the method works https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch
```suggestion
    switch (sortBy) {
```